hr_shift/models/shift_planning.py

```diff
@@ -0,0 +1,481 @@
+        )
+        return action
+
+
+class ShiftPlanningShift(models.Model):
+    _name = "hr.shift.planning.shift"
+    _description = "Shift of the week for a given employee"
+
+    planning_id = fields.Many2one(comodel_name="hr.shift.planning", ondelete="cascade")
+    employee_id = fields.Many2one(comodel_name="hr.employee", ondelete="cascade")
+    image_128 = fields.Image(related="employee_id.image_128")
+    template_id = fields.Many2one(
+        comodel_name="hr.shift.template", group_expand="_group_expand_template_id"
+    )
+    color = fields.Integer(related="template_id.color")
+    line_ids = fields.One2many(
+        comodel_name="hr.shift.planning.line", inverse_name="shift_id"
+    )
+    lines_data = fields.Serialized(default={}, compute="_compute_lines_data")
+    state = fields.Selection(
+        selection=[
+            ("available", "Fully available"),
+            ("partial", "Partially available"),
+            ("unavailable", "Unavailable"),
+        ],
+    )
+    reviewed = fields.Boolean(
+        help="For unavailable shifts that we don't want to be listed as issued",
+        compute="_compute_reviewed",
+        inverse="_inverse_reviewed",
+        readonly=False,
+    )
+
+    _sql_constraints = [
+        (
+            "unique_planning_employee",
+            "unique(planning_id,employee_id)",
+            "You can't assign an employee twice to the same plan!",
+        )
+    ]
+
+    @api.model
+    def _group_expand_template_id(self, templates, domain, order):
+        return self.env["hr.shift.template"].search([])
+
+    @api.depends("line_ids")
+    def _compute_lines_data(self):
+        for shift in self.filtered("line_ids"):
+            shift.lines_data = {
+                line.id: {
+                    "day": dict(WEEK_DAYS_SELECTION).get(line.day_number),
+                    "template": line.template_id.name,
+                    "state": line.state,
+                    "color": line.color,
+                }
+                for line in shift.line_ids
+            }
+
+    @api.depends("line_ids.reviewed")
+    def _compute_reviewed(self):
+        for shift in self:
+            shift.reviewed = all(shift.line_ids.mapped("reviewed"))
+
+    def _inverse_reviewed(self):
+        for shift in self:
+            shift.line_ids.reviewed = shift.reviewed
+
+    def _generate_shift_lines(self):
+        for shift in self:
+            dates = shift.template_id._explode_date_range(
+                shift.planning_id.start_date, shift.planning_id.end_date
+            )
+            shift_lines = []
+            for shift_date in dates:
+                shift_lines.append(
+                    {
+                        "shift_id": shift.id,
+                        "day_number": str(shift_date["weekday"]),
+                    }
+                )
+            lines = shift.line_ids.create(shift_lines)
+            lines._compute_state()
+            lines._compute_template_id()
```

Comment:
```suggestion
                shift_lines.append(
                    {
                        "shift_id": shift.id,
                        "template_id": shift.template_id.id,
                        "day_number": str(shift_date["weekday"]),
                    }
                )
            shift.line_ids.create(shift_lines)
```
I believe that by defining the template (_template_id_) in the lines, it will no longer be necessary to call the `_compute_template_id()` method or the `_compute_state()` method.